### ISSUE TITLE
git diff doesn't work for files inside folders

### ISSUE THREAD

**Issue by brukau** [jupyterlab/jupyterlab-git#391]
Seems like path is not set correctly on /gitdiff POST request

in my case project structure is like
project->folder->notebook.ipynb

POST request to /gitdiff includes
`"file_path": "folder/notebook.ipynb" `
and it fails with 422 reply:

`HTTP 422: Unprocessable Entity (Invalid notebook: /home/jovyan/folder/notebook.ipynb)`
("project" is missing in the path)

instead of 
`"file_path": "project/folder/notebook.ipynb"`

**Comment by jaipreet-s** [comment]
Hmm I have a file in a folder `somefolder/notebook.ipynb` and that works fine. 
Checking for `someproject/somefolder/notebook.ipynb` now

**Comment by brukau** [comment]
That's how I can reproduce it

start jupyter lab and install extension in docker container:
```
docker run --rm -it -p 8888:8888 jupyter/minimal-notebook:7a3e968dd212 bash
pip install jupyterlab-git
jupyter labextension install @jupyterlab/git
jupyter serverextension enable --py jupyterlab_git
jupyter-lab
```

open jupyterlab http://localhost:8888/?token=....

clone for example this repo https://github.com/microsoft/recommenders using clone button
open for example *recommeners/notebooks/03_evaluate/evaluation.ipynb*
change some cell
you can see the diff with nbdime
but when I tried to click on diff icon it fails
![image](https://user-images.githubusercontent.com/15345943/62685337-c451cd00-b9c2-11e9-85d7-379718725cd7.png)

**Comment by brukau** [comment]
also I got the same errors in the latest (1.0.4) jupyterlab

**Comment by jaipreet-s** [comment]
@brukau : Thanks for the repro steps. I'm able to repro this inside the Docker container. Let me take a look at whats going on.

**Comment by jaipreet-s** [comment]
This is happening because in nbdime we are always using the path relative to the server root, and in some cases the git extension does not pass that. In the Git extension, the information we have is the Git repo root and  the path of the file relative to the root. 

Possible solutions:

1. Update nbdime to accept absolute paths, and the Git extension to always pass absolute paths
    - This is good because it completely takes us out of the business of issues with relative and nested paths
    - https://github.com/jupyter/nbdime/pull/494 created for discussion

2. Add a server endpoint in jupyterlab-git to return the server root similar to https://github.com/jupyter/nbdime/blob/master/nbdime/webapp/nb_server_extension.py#L90-L91
    - `PathExt.join(PathExt.relative(server_root, git_root), git_relative_path)` will give the the path of the file relative to the server root then

3. Use the current file browser path and prepend that to the file path before calling the API. 
    - This does not work in the case where the file-browser is at `repo/subfolder` and the file is at `subfolder/file`, since joining these 2 paths will produce `repo/subfolder/subfolder/file`

**Comment by jaipreet-s** [comment]
Hi @brukau ,

I've pushed a fix to a branch(#395) and tested this out from my side.Would you like to test this from and seeing if this works in your setup

```
docker run --rm -it -p 8888:8888 jupyter/minimal-notebook:7a3e968dd212 bash
git clone https://github.com/jupyterlab/jupyterlab-git.git --branch pathfix2
cd jupyterlab-git/
pip install .
jupyter labextension install .
jupyter serverextension enable --py jupyterlab_git
cd ..
jupyter-lab
```

**Comment by brukau** [comment]
hi @jaipreet-s 

seems working, well done!
